topic_tools/src/tool_base_node.cpp

```diff
@@ -30,6 +30,14 @@ ToolBaseNode::ToolBaseNode(const std::string & node_name, const rclcpp::NodeOpti
 void ToolBaseNode::make_subscribe_unsubscribe_decisions()
 {
   if (auto source_info = try_discover_source()) {
+    // publisher exists already but needs changing if output_topic_ changes
+    if (
+      pub_ &&
+      pub_->get_topic_name() != get_node_topics_interface()->resolve_topic_name(output_topic_))
+    {
+      pub_.reset();
+    }
```

Comment:
Nitpick style suggestion.
```suggestion
    if (pub_ &&
      pub_->get_topic_name() != get_node_topics_interface()->resolve_topic_name(output_topic_))
    {
      pub_.reset();
    }
```

Reply:
Accepted